bee-node/src/plugins/mqtt/broker.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{plugins::mqtt::handlers::*, storage::StorageBackend};
+
+use bee_runtime::{node::Node, worker::Worker};
+
+use async_trait::async_trait;
+
+use librumqttd::{LinkRx, LinkTx};
+use log::*;
+
+use std::convert::Infallible;
+
+pub struct MqttBrokerConfig {
```

Comment:
maybe "Config" is not the best name for this type as it doesn't contain any configuration data, but only channel halfs. Might still be from what I introduced back then. (Feel free to change if so :smile: ).

Reply:
I see why it's done like this, but it's still confusing. Can we not do what is done when creating this struct within `start` down below, and then return a tuple `(Self, ...channel_halfs...)`. That would mean that the `LinkTx, LinkRx` are created within `start`, and also we would have`type Config = ()` until we are able to properly configure the broker (with some of those constants from the other module)

Reply:
Agree with you that "Config" is not the best name for this type as it doesn't contain any configuration data, but only channel halfs. So I would rename this too.

Regarding your second comment:
If I understood correctly, you would like to move [following code](https://github.com/rufsam/bee/blob/chrysalis-pt-2/bee-node/src/plugins/mqtt/mod.rs#L95-L188) to the `start()` function right?
Also, instead of passing [here ](https://github.com/rufsam/bee/blob/chrysalis-pt-2/bee-node/src/plugins/mqtt/mod.rs#L208) the channel halfs, we could pass [that config](https://github.com/rufsam/bee/blob/chrysalis-pt-2/bee-node/src/plugins/mqtt/mod.rs#L49),